test/scripts/openshift-ci/deploy.serverless.sh

```diff
@@ -98,6 +98,12 @@ spec:
       enabled: true
 EOF
 
+# Apparently, as part of KNative installation, deployments can be restarted because
+# of configuration changes, leading to waitpodready to fail sometimes.
+# Let's sleep 2minutes to let the KNative operator to stabilize the installation before
+# checking for the readiness of KNative stack.
+sleep 120
```

Comment:
why don't use a more accurate way to check if the pod is ready?
something like `oc wait` or [this kind of script](https://github.com/opendatahub-io/caikit-tgis-serving/blob/main/demo/kserve/scripts/utils.sh#L60)
This is [an example](https://github.com/opendatahub-io/caikit-tgis-serving/blob/main/demo/kserve/scripts/test/grpc-call.sh#L13-L14)

Reply:
The `waitpodready` function that is used on the following lines is using `oc wait`.

The problem is that `oc wait` fails if the deployment is restarted or a new version is rolled out, while waiting. So, it is a race condition; i.e. if `oc wait` is invoked before a new rollout, it will be watching the older pods and it will fail because those older pods will be terminated (`oc wait` will complain that the older pods can no longer be found).

The sleep is to try to prevent such race condition.

Reply:
So, as mentioned in the comment, the sleep is to let the KNative operator to stabilize, and be done with new rollouts. It is not the goal to wait for the pods to become ready.